### ISSUE TITLE
Inconsistency between the paper and the code implementation

### ISSUE THREAD

**Issue by HEX-QWQ** [raymin0223/mixture_of_recursions#10]
Hi raymin0223 ,
Thank you for sharing your excellent work and open-sourcing the code!  

While learning both the paper and the implementation, I came across a potential inconsistency.

In Appendix A.2. Routing Strategy, it is mentioned:
> The auxiliary router is an additional lightweight network trained jointly but used **only during inference**; it predicts whether a token will be among the top-k selection.

However, in `expert_choice_router.py` at line 110, the auxiliary router `self.mlp_router` is defined, but it is only used when the condition `self.training` at line 191 is met. This condition indicates that the model is in the training phase rather than the inference phase.

Is the absence of `self.mlp_router` in inference intentional, or is this a partial implementation?

Thank you in advance for your time and explanation!

**Comment by raymin0223** [comment]
Hi @HEX-QWQ , sorry for the late response.

I guess this also stems from the absence of inference code for MoR (#9).
[expert_choice_router.py](https://github.com/raymin0223/mixture_of_recursions/blob/main/model/mor_model/expert_choice_router.py) is only for "training" phase, sorry.
 
Auxiliary router (defined as `self.mlp_router`) should be trained together with `self.mor_router`, if we want to use auxiliary router method in expert-choice.
And during inference, we will remove `mor_router` and instead use `mlp_router` for sure. 

We didn't actually implement auxiliary router case for inference, since our final method is using auxiliary loss and aux routers show worse top-k prediction accuracy.

**Comment by HEX-QWQ** [comment]
Hi @raymin0223,

Thank you so much for the explanation,, it clears my confusion. thank you for sharing your excellent work and open-sourcing the code again!

Best regards,
